pkg/raft/raft.go

```diff
@@ -1035,6 +1035,8 @@ func (r *raft) campaign(t CampaignType) {
 		// better safe than sorry.
 		r.logger.Warningf("%x is unpromotable; campaign() should have been called", r.id)
 	}
+	// If already a (pre)candidate, the previous election attempt failed.
+	hung := r.state != StateFollower
```

Comment:
Can `r.state == StateLeader` here? By looking at the callers no, but maybe worth noting it here since you rely on it to conclude that we're in a (pre)candidate state.